fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java

```diff
@@ -143,13 +145,15 @@ private void auditAfterExec(String origStmt, StatementBase parsedStmt, Data.PQue
         
         ctx.getAuditEventBuilder().setFeIp(FrontendOptions.getLocalHostAddress());
 
-        // We put origin query stmt at the end of audit log, for parsing the log more convenient.
+        String sql;
         if (!ctx.getState().isQuery() && (parsedStmt != null && parsedStmt.needAuditEncryption())) {
-            ctx.getAuditEventBuilder().setStmt(parsedStmt.toSql());
+            sql = parsedStmt.toSql();
         } else {
-            ctx.getAuditEventBuilder().setStmt(origStmt);
+            sql = origStmt;
         }
-        
+        ctx.getAuditEventBuilder().setSqlHash(DigestUtils.md5Hex(sql));
```

Comment:
calc md5 of SQL once and save it in connect ctx, so that we can reuse it.

Reply:
Ok, let me think about this